app/src/main/java/com/android/sample/ui/createRecipe/RecipeListInstructionsScreen.kt

```diff
@@ -0,0 +1,270 @@
+      }
+}
+
+@Composable
+fun InstructionValue(
+    createRecipeViewModel: CreateRecipeViewModel,
+    index: Int,
+    time: String?,
+    onClick: (Int) -> Unit
+) {
+  val officialStep = index + 1
+  Card(
+      modifier =
+          Modifier.testTag(INSTRUCTION_LIST_ITEM)
+              .fillMaxWidth(0.9f)
+              .padding(horizontal = MEDIUM_PADDING.dp)
+              .padding(vertical = REALLY_SMALL_PADDING.dp)
+              .clickable(onClick = { onClick(index) })
+              .shadow(
+                  elevation = CARD_SHADOW_ELEVATION.dp,
+                  shape = RoundedCornerShape(CARD_BORDER_ROUND.dp),
+              ),
+      colors =
+          CardDefaults.cardColors(
+              containerColor = MaterialTheme.colorScheme.background, // Background color of the card
+              contentColor = MaterialTheme.colorScheme.onPrimary // Content color of the card
+              ),
+      shape = RoundedCornerShape(CARD_CORNER_RADIUS.dp)) {
+        Column(modifier = Modifier.fillMaxWidth().padding(SMALL_PADDING.dp)) {
+          // Date and Status Row
+          Row(
+              modifier = Modifier.fillMaxWidth(ROW_SIZE).testTag(RECIPE_LIST_ITEM_THUMBNAIL),
+              horizontalArrangement = Arrangement.SpaceBetween,
+              verticalAlignment = Alignment.CenterVertically) {
+                Image(
+                    painter =
+                        painterResource(
+                            if (createRecipeViewModel.getIcon(index) != null) {
+                              createRecipeViewModel.getIcon(index)!!.iconResId
+                            } else {
+                              R.drawable.fire
+                            }),
+                    contentDescription = "Icon",
+                    modifier = Modifier.size(ICON_SIZE.dp).testTag(RECIPE_LIST_INSTRUCTION_ICON))
+
+                Column(modifier = Modifier.testTag(INSTRUCTION_TEXT_SPACE)) {
+                  Text(
+                      modifier = Modifier.testTag(INSTRUCTION_TEXT_IN_CARD),
+                      text =
+                          "${stringResource(R.string.RecipeListInstructionsScreen_Step)} $officialStep",
+                      style = MaterialTheme.typography.bodyMedium,
+                      fontWeight = FontWeight.Bold)
+
+                  if (!time.isNullOrBlank()) {
+                    Text(
+                        text =
+                            "$time ${stringResource(R.string.RecipeListInstructionsScreen_Minutes)}",
+                        style = MaterialTheme.typography.bodySmall,
+                        modifier = Modifier.testTag(INSTRUCTION_TIME))
+                  }
+                }
+                Icon(
+                    imageVector = Icons.Default.ModeEdit,
+                    contentDescription = "Edit",
+                    modifier = Modifier.size(ICON_SIZE.dp).testTag(EDIT_INSTRUCTION_ICON))
+              }
+        }
+      }
+}
+
+/**
```

Comment:
That's a fantastic idea! I really like the approach you're taking here, nicely done!

Reply:
Thank you !!